src/main/java/com/example/petbuddybackend/entity/user/Caretaker.java

```diff
@@ -45,12 +45,16 @@ public class Caretaker {
     private Integer numberOfRatings;
 
     @Basic(fetch = FetchType.LAZY)
-    @Formula("(SELECT AVG(r.rating) " +
+    @Formula("(SELECT COALESCE(AVG(r.rating), 0.0) " +
             " FROM Rating r " +
             " JOIN Care c ON r.care_id = c.id " +
             " WHERE c.caretaker_email = email)")
     private Float avgRating;
```

Comment:
Właśnie `avgRating` może być null i wydaje mi się że frontend już to handluje

Reply:
przeciez mowiles zeby 0 zwracac

Reply:
Tak, ale dla średniej ogólnej raczej. Tutaj dla konkretnego opiekuna bym zostawił